src/portable/portablectl.c

```diff
@@ -384,24 +418,31 @@ static int inspect_image(int argc, char *argv[], void *userdata) {
                                 fflush(stdout);
                                 nl = true;
                         } else {
-                                _cleanup_free_ char *pretty_portable = NULL, *pretty_os = NULL, *sysext_level = NULL,
-                                        *sysext_id = NULL, *sysext_version_id = NULL, *sysext_scope = NULL, *portable_prefixes = NULL,
+                                _cleanup_free_ char *pretty_portable = NULL, *pretty_os = NULL, *extension_level = NULL,
+                                        *extension_id = NULL, *extension_version_id = NULL, *extension_scope = NULL, *portable_prefixes = NULL,
                                         *id = NULL, *version_id = NULL, *image_id = NULL, *image_version = NULL, *build_id = NULL;
+                                _cleanup_free_ char *temp_level_check = NULL;
                                 _cleanup_fclose_ FILE *f = NULL;
+                                ImageClass class = IMAGE_SYSEXT;
 
                                 f = fmemopen_unlocked((void*) data, sz, "r");
                                 if (!f)
                                         return log_error_errno(errno, "Failed to open extension-release buffer: %m");
 
+                                r = parse_env_file(f, name, "CONFEXT_LEVEL", &temp_level_check);
+                                if (!isempty(temp_level_check)) {
+                                        class = IMAGE_CONFEXT;
+                                }
+
                                 r = parse_env_file(f, name,
-                                                   "SYSEXT_ID", &sysext_id,
-                                                   "SYSEXT_VERSION_ID", &sysext_version_id,
-                                                   "SYSEXT_BUILD_ID", &build_id,
-                                                   "SYSEXT_IMAGE_ID", &image_id,
-                                                   "SYSEXT_IMAGE_VERSION", &image_version,
-                                                   "SYSEXT_PRETTY_NAME", &pretty_os,
-                                                   "SYSEXT_SCOPE", &sysext_scope,
-                                                   "SYSEXT_LEVEL", &sysext_level,
+                                                   image_class_info[class].id, &extension_id,
```

Comment:
this shouldn't be necessary just get both sets at the same time, one will be filled and one will be just NULLs